alti/lib/profile_helpers.py

```diff
@@ -82,9 +100,28 @@ def _create_profile(coordinates, z_values, output_to_json):
     return profile
 
 
-def _smart_fill(coordinates, nb_points):
+def _prepare_number_of_points_max_per_segment(coordinates, nb_point_total):
+
+    nb_points_segments = list()
+    distances = []
+    distances_squareform = squareform(pdist(coordinates))
```

Comment:
Hey I don't mean to be intrusive, I am just interested in the sampling method for the profile.

That seems like a lot of extra unnecessary work no? you are computing n^2 distances when you only need the distance between each point (n-1). Wouldn't "Pythagoras Theorem" without scipy be a better alternative? I think you could avoid a bunch of loops and keep the code much tighter. Something like:

```python
import math
def square_dist(a, b):
    return math.sqrt(math.pow(a[0] - b[0], 2) + math.pow(a[1] - b[1], 2))
distances = []
total_disance = 0
# If python 2
for i in xrange(0, len(coordinates) - 1):
    dist = square_dist(coordinates[i], coordinates[i+1])
    distances.append(dist)
    total_distance += dist
```

Reply:
That's true. I simply reused the existing method, thinking there was a reason why it was done this way.